types.go

```diff
@@ -1798,31 +1799,31 @@ func (m *Message) GetData() []byte {
 // Accept notifies the server that the message has been
 // accepted and does not require redelivery.
 func (m *Message) Accept(ctx context.Context) error {
-	defer m.done()
```

Comment:
Unfortunately we now leak the channel in mode one as `m.shouldSendDisposition()` returns false.  Ideally we'd never create the channel for mode one as we don't need to track the message being settled.